php/class-ad-layers.php

```diff
@@ -245,11 +245,14 @@ public function set_active_ad_layer() {
 						continue;
 					}
 
-					// Check taxonomies
+					// Check taxonomies.
+					$taxonomy_match = false;
 					if ( ! empty( $taxonomy_terms ) ) {
 						foreach ( $taxonomy_terms as $taxonomy => $terms ) {
 							if ( has_term( $terms, $taxonomy ) ) {
 								$taxonomy_match = true;
+							} else {
+								$taxonomy_match = false;
```

Comment:
This appears to be a breaking change since themes might be relying on Ad Layers to match for only a single taxonomy.  The code is sound, but I just wanted to mention the concern for backwards compatibilty.  Maybe an Ad Layers expert can weigh in.